generators/app/templates/src/containers/Profile/profile.container.js

```diff
@@ -88,44 +95,67 @@ export class Profile extends Component {
       e.preventDefault();
       let node;
       let nextAction;
-      const updatedFormField = await Promise.all(
-        this.state.formFields.map(async field => {
-          if (field.updated) {
-            node = data.user[field.property];
-            nextAction = 'update';
+      let updatedFields = [];
 
-            if (field.blankNode) {
-              node = data[field.nodeParentUri][field.blankNode];
-            }
+      this.setState({ isLoading: true });
+      /*
+       * Solid server has an issue on concurrent updates,
+       * so to fix this we have to await one change, and only when it is done, 
+       * fire the next field update.
+       * more info about the issue: https://github.com/solid/node-solid-server/issues/1106
+      */
+      for await (const field of this.state.formFields) {
+        if (field.updated) {
+          node = data.user[field.property];
+          nextAction = 'update';
 
-            if (field.action === 'update') {
-              await node.set(field.value);
-            } else if (field.action === 'create') {
-              await node.add(field.value);
-            } else {
-              await node.delete();
-              nextAction = 'create';
-            }
+          if (field.blankNode) {
+            node = data[field.nodeParentUri][field.blankNode];
+          }
+
+          if (field.action === 'update') {
+            await node.set(field.value);
+          } else if (field.action === 'create') {
+            await node.add(field.value);
```

Comment:
Create and update can both be done by `set`. So you probably want `if delete delete ele set`, or just to reduce your actions to not include create.

Reply:
@RubenVerborgh thank you, This is done